src/test/java/seedu/duke/command/DeleteCommandTest.java

```diff
@@ -0,0 +1,78 @@
+package seedu.duke.command;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import seedu.duke.financial.FinancialList;
+import seedu.duke.financial.Expense;
+import seedu.duke.financial.Income;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/**
+ * Test class for the DeleteCommand.
+ * This class contains unit tests to verify the functionality of the DeleteCommand
+ * when removing an entry from the financial list.
+ */
+class DeleteCommandTest {
+
+    private final ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+    private final PrintStream originalOut = System.out;
+    private FinancialList financialList;
+    private DeleteCommand deleteCommand;
+
+    /**
+     * Set up the test environment before each test.
+     * Initializes the FinancialList and redirects System.out to capture console output.
+     */
+    @BeforeEach
+    void setUp() {
+        financialList = new FinancialList();
+        System.setOut(new PrintStream(outputStream));
+    }
+
+    /**
+     * Clean up the test environment after each test.
+     * Restores the original System.out PrintStream.
+     */
+    @AfterEach
+    void tearDown() {
+        System.setOut(originalOut);
+    }
+
+    /**
+     * Test the execute method of DeleteCommand.
+     * Verifies that an entry is deleted from the financial list when a valid index is given.
+     */
+    @Test
+    void execute_deleteEntry_expectEntryRemoved() {
```

Comment:
Perhaps you can add a check to verify that the correct task was deleted (e.g. check the description of the remaining task) ?

Reply:
Sounds good, ill add it in.